Append the branch name to non prod pipelines by default

marksliva opened the pull request:
So that machines with the same user do not accidentally clobber
existing pipeline.

Co-authored-by: Amil Khanzada @amilkh 

## Here are some reminders after you submit the pull request
- [ ] cherry pick to 6X_STABLE and 5X?

amilkh commented:
good point! looks like we have to be careful here:
```
    # if generating a dev pipeline but didn't specify an output,
    # don't overwrite the master pipeline
    if args.pipeline_target != 'prod' and not output_path_is_set:
        pipeline_file_suffix = args.user
        if args.use_branch:
            pipeline_file_suffix = suggested_git_branch()
        default_dev_output_filename = 'gpdb-' + args.pipeline_target + '-' + pipeline_file_suffix + '.yml'
        args.output_filepath = os.path.join(PIPELINES_DIR, default_dev_output_filename)
```

marksliva commented:
> Only one of `-b`, `-u`, or `-o` can be specified. Please verify that enabling `-b` by default does not break `-u` or `-o` if one of them is specified. Otherwise, looks good.

@slari-pivotal Addressed your comment in this fixup commit https://github.com/greenplum-db/gpdb/pull/7971/commits/12158aab504f892c0497a3835490e88e5a49e479

amilkh commented:
cherry-picked to 6X_STABLE also
